cmd/fluxctl/sync.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/weaveworks/flux/git"
+	"github.com/weaveworks/flux/update"
+)
+
+type syncOpts struct {
+	*rootOpts
+}
+
+func newSync(parent *rootOpts) *syncOpts {
+	return &syncOpts{rootOpts: parent}
+}
+
+func (opts *syncOpts) Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sync",
+		Short: "synchronize the cluster with the git repository, now",
+		RunE:  opts.RunE,
+	}
+	return cmd
+}
+
+func (opts *syncOpts) RunE(cmd *cobra.Command, args []string) error {
+	if len(args) > 0 {
+		return errorWantedNoArgs
+	}
+
+	ctx := context.Background()
+
+	gitConfig, err := opts.API.GitRepoConfig(ctx, false)
+	if err != nil {
+		return err
+	}
+
+	switch gitConfig.Status {
+	case git.RepoNoConfig:
+		return fmt.Errorf("no git repository is configured")
+	case git.RepoReady:
+		break
+	default:
+		return fmt.Errorf("git repository %s is not ready to sync (status: %s)", gitConfig.Remote.URL, string(gitConfig.Status))
+	}
+
+	fmt.Fprintf(cmd.OutOrStderr(), "Synchronizing with %s\n", gitConfig.Remote.URL)
+
+	updateSpec := update.Spec{
+		Type: update.Sync,
+		Spec: update.ManualSync{},
+	}
+	jobID, err := opts.API.UpdateManifests(ctx, updateSpec)
```

Comment:
Should this be something other than an `UpdateManifests` call? No manifests are updated in the repo so this feels like existing-API abuse.

Reply:
I considered adding another API method just for the sync call, but since it had almost everything in common with updates (goes in the job queue, result is a revision, which you then wait for), I decided to use the same mechanism. There are some small mismatches, aside from the name -- there's no cause, or `update.Result`, associated with the sync.

And, to be honest, I felt dread at the prospect of adding an API method.

Arguably, this method should really be called `SubmitJob`. Maybe in the next API version we can finesse it a little.